fsspec/implementations/zip.py

```diff
@@ -133,3 +141,16 @@ def _open(
 
     def ukey(self, path):
         return tokenize(path, self.fo, self.protocol)
+
+    def _all_dirnames(self, paths):
+        """ Returns *all* directory names for each path in paths, including intermediate ones.
+
+        Parameters
+        ----------
+        paths: Iterable of path strings
+        """
+        if len(paths) == 0:
+            return set()
+
+        dirnames = {os.path.dirname(path) for path in paths} - {f"{self.root_marker}"}
```

Comment:
`self.root_marker` is a string anyway, no need for format.

Reply:
I got a little too carried away on `f"{}"-syntax`. It is fixed...